### DISCUSSION TITLE
Deleting current skin while skin editor is open will immediately create a new copy of the default

### DISCUSSION THREAD

**Discussion by sona-tau** [ppy/osu#22108] [Q&A]
### Type

Game behaviour

### Bug description

Deleting a modified skin while the skin editor is open creates a new skin called: osu! "argon" (2022) (modified) (#) (Unkown). This skin is practically the same as the normal argon skin. It'll create a new copy incrementing the (#) part every time you delete a modified skin while in the skin editor. The modified skin is undeletable though.

System: 
- Void Linux x86_64
- X11
- kernel 6.0.11_1

_PS. This is the first issue I have ever made, if I make any mistakes feel free to tell me and I'll gladly fix them!_

### Screenshots or videos

https://user-images.githubusercontent.com/65361987/207882185-fc46f6f8-2308-44f7-b429-5ae9c5a705ab.mp4

### Version

2022.1208.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10237662/database.log)
[input.log](https://github.com/ppy/osu/files/10237663/input.log)
[network.log](https://github.com/ppy/osu/files/10237664/network.log)
[performance.log](https://github.com/ppy/osu/files/10237665/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10237666/runtime.log)

**Comment by peppy** [comment]
This is *kind of* intentional. When the skin editor is open, it will always create a new copy if the current skin is the default, to ensure you can edit it. What is happening here is precisely this – you're deleting the active skin and the game is reverting to default, then the editor is creating a new copy for you to edit.

I'm not sure how we should handle this. Maybe a better approach would be to have the skin editor just not work when a default is selected, and have a "Create copy" button in the centre of the screen?

**Reply by bdach** [reply]
@ppy/area-client prompt for opinions (also testing if pinging the team works)

**Reply by frenzibyte** [reply]
I personally get annoyed when opening skin editor in default skin and suddenly a duplicate is made even though I don't intend on editing anything (my database stores a handful of duplicates right now).

It would be optimal (at least for me) to allow accessing the skin editor but in "read only" mode, but if that's too much effort, then just disabling access to skin editor content and potentially adding a button to select latest duplicated skin next to creating another copy would be good.

**Reply by Joehuu** [reply]
> have the skin editor just not work when a default is selected, and have a "Create copy" button in the centre of the screen

I agree to disable it as a start, but I believe the button shouldn't be obscuring the content and should be somewhere at the top or inside "File" as creating copies can potentially be used on non-default skins.

> allow accessing the skin editor but in "read only" mode

This is also the ideal behavior I want (i.e. seeing skin elements and their details, but not actually editing), but for another day I guess.

> potentially adding a button to select latest duplicated skin

Not sure about linking duplicate skins, a skin can drastically change to a point it doesn't even look like the copy.

**Reply by peppy** [reply]
> I agree to disable it as a start, but I believe the button shouldn't be obscuring the content and should be somewhere at the top or inside "File" as creating copies can potentially be used on non-default skins.

Probably both should exist. I imagine for the default skin, the editor main area will be dimmed with a message in the middle above the dim "This skin is currently read only [Create a copy]"
